src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -448,4 +461,275 @@ def parse_chat(chat_str, images: List = None, valid_roles: List[str] = None):
+        # If this config file isn't set up, it will return None.
+        return get_workspace_triad_from_local()
+
+
+def list_deployment_connections(
+    subscription_id=None,
+    resource_group_name=None,
+    workspace_name=None,
+    connection="",
+):
+    try:
+        # Do not support dynamic list if azure packages are not installed.
+        from azure.mgmt.cognitiveservices import CognitiveServicesManagementClient
+
+        from promptflow.azure.operations._arm_connection_operations import (
+            ArmConnectionOperations,
+            OpenURLFailedUserError,
+        )
+    except ImportError:
+        return None
+
+    # Do not support dynamic list if the workspace triple is set in the local.
+    if not subscription_id or not resource_group_name or not workspace_name:
+        return None
+
+    try:
+        credential = _get_credential()
+        try:
+            # Currently, the param 'connection' is str, not AzureOpenAIConnection type.
+            conn = ArmConnectionOperations._build_connection_dict(
+                name=connection,
+                subscription_id=subscription_id,
+                resource_group_name=resource_group_name,
+                workspace_name=workspace_name,
+                credential=credential,
+            )
+            resource_id = conn.get("value").get("resource_id", "")
+            if not resource_id:
+                return None
+            conn_sub, conn_rg, conn_account = _parse_resource_id(resource_id)
+        except OpenURLFailedUserError:
+            return None
+        except ListDeploymentsError as e:
+            raise e
+        except Exception as e:
+            msg = f"Parsing connection with exception: {e}"
+            raise ListDeploymentsError(msg=msg) from e
+
+        client = CognitiveServicesManagementClient(
+            credential=credential,
+            subscription_id=conn_sub,
+        )
+        return client.deployments.list(
+            resource_group_name=conn_rg,
+            account_name=conn_account,
+        )
+    except Exception as e:
+        if hasattr(e, "status_code") and e.status_code == 403:
+            msg = f"Failed to list deployments due to permission issue: {e}"
+            raise ListDeploymentsError(msg=msg) from e
+        else:
+            msg = f"Failed to list deployments with exception: {e}"
+            raise ListDeploymentsError(msg=msg) from e
+
+
+def refine_extra_fields_not_permitted_error(connection, deployment_name, model):
+    tsg = (
+        "Please kindly avoid using vision model in LLM tool, "
+        "because vision model cannot work with some chat api parameters. "
+        "You can change to use tool 'Azure OpenAI GPT-4 Turbo with Vision' "
+        "or 'OpenAI GPT-4V' for vision model."
+    )
+    try:
+        if isinstance(connection, AzureOpenAIConnection):
+            subscription_id, resource_group, workspace_name = get_workspace_triad()
```

Comment:
are we alwasy in connection provider = azure here?

Reply:
remove it, this error message refine is used for LLM tool